source/ice_api.c

```diff
@@ -0,0 +1,1676 @@
+                if( Ice_IsSameIpAddress( &stunAddress,
+                                         &pIceAgent->remoteCandidates[ i ].ipAddress.ipAddress,
+                                         true ) == true )
+                {
+                    *ppCandidate = &pIceAgent->remoteCandidates[ i ];
+                    foundCandidate = true;
+                    break;
+                }
+            }
+        }
+    }
+
+    return foundCandidate;
+}
+/*------------------------------------------------------------------------------------------------------------------*/
+
+/* Ice_FindCandidatePairWithLocalAndRemoteAddr - This API is called internally to find a candidate pair given the local and remote IP addresses. */
+
+static bool Ice_FindCandidatePairWithLocalAndRemoteAddr( IceAgent_t * pIceAgent,
+                                                         IceIPAddress_t * pSrcAddr,
+                                                         IceIPAddress_t * pRemoteAddr,
+                                                         IceCandidatePair_t ** ppCandidatePair )
+{
+    int i;
+    int candidatePairCount;
+    IceCandidatePair_t iceCandidatePair;
+    bool found = false;
+
+    if( ( pIceAgent == NULL ) ||
+        ( pSrcAddr == NULL ) ||
+        ( pRemoteAddr == NULL ) ||
+        ( ppCandidatePair == NULL ) )
+    {
+        found = false;
+    }
+    else
+    {
+        candidatePairCount = Ice_GetValidCandidatePairCount( pIceAgent );
+
+        for( i = 0; i < candidatePairCount; i++ )
+        {
+            iceCandidatePair = pIceAgent->iceCandidatePairs[ i ];
+
+            if( ( Ice_IsSameIpAddress( &( iceCandidatePair.pLocal->ipAddress.ipAddress ),
+                                       &( pSrcAddr->ipAddress ),
+                                       true ) ) && ( Ice_IsSameIpAddress( &( iceCandidatePair.pRemote->ipAddress.ipAddress ),
+                                                                          &( pRemoteAddr->ipAddress ),
+                                                                          true ) ) )
+            {
+                *ppCandidatePair = &( pIceAgent->iceCandidatePairs[ i ] );
+                found = true;
+            }
+        }
+    }
+
+    return found;
+}
+/*------------------------------------------------------------------------------------------------------------------*/
+
+/* Ice_IsSameIpAddress - This API is called internally to check if two IPAddresses are same. */
+
+static bool Ice_IsSameIpAddress( StunAttributeAddress_t * pAddr1,
+                                 StunAttributeAddress_t * pAddr2,
+                                 bool checkPort )
+{
+    bool ret;
+    uint32_t addrLen;
+
+    if( ( pAddr1 == NULL ) || ( pAddr2 == NULL ) )
+    {
+        ret = false;
+    }
+    else
+    {
+        addrLen = ICE_IS_IPV4_ADDR( *pAddr1 ) ? STUN_IPV4_ADDRESS_SIZE : STUN_IPV6_ADDRESS_SIZE;
+
+        ret = ( pAddr1->family == pAddr2->family && memcmp( pAddr1->address,
```

Comment:
Can we decouple it into multiple line? to be more readable.

Reply:
updated.